src/event_listener.rs

```diff
@@ -137,6 +146,34 @@ pub enum FlushReason {
     CatchUpAfterErrorRecovery = 14,
 }
 
+impl Drop for FlushJobInfo<'_> {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::rocksdb_table_properties_destroy(self.table_properties.as_ptr());
+        }
+    }
+}
+
+impl<'a> FlushJobInfo<'a> {
+    pub fn get_user_collected_property(&self, key: impl CStrLike) -> Option<CString> {
```

Comment:
`CStrLike` is a trait provided by `rust-rocksdb` - it allows the user to use any of `&str` | `&String` | `&CStr` | `&CString` when calling the API. Unfortunately lifetime requirements mean that this doesn't work great as a return type for `name()` elsewhere.